Cannot find module 'ajv/dist/compile/codegen'

### Current Behavior

When i create a new nestjs app and run the command `npx nx serve api` i get the error 
```
[webpack-cli] Error: Cannot find module 'ajv/dist/compile/codegen'
...
Warning: run-commands command "webpack-cli build --node-env=production" exited with non-zero status code

```


### Expected Behavior

The expected behavior is that the development server should run

### GitHub Repo

https://github.com/sagarPakhrin/samuha

### Steps to Reproduce

1. npx create-nx-workspace my-workspace --preset=nest
2. Follow the prompt and name the app api
3. cd my-workspace
4. npx nx serve api


### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node   : 20.9.0
   OS     : darwin-arm64
   npm    : 10.1.0

   nx                 : 18.0.0
   @nx/js             : 18.0.0
   @nx/jest           : 18.0.0
   @nx/linter         : 18.0.0
   @nx/eslint         : 18.0.0
   @nx/workspace      : 18.0.0
   @nx/cypress        : 18.0.0
   @nx/devkit         : 18.0.0
   @nx/eslint-plugin  : 18.0.0
   @nx/nest           : 18.0.0
   @nx/node           : 18.0.0
   @nx/react          : 18.0.0
   @nrwl/tao          : 18.0.0
   @nx/vite           : 18.0.0
   @nx/web            : 18.0.0
   @nx/webpack        : 18.0.0
   typescript         : 5.3.3
```


### Failure Logs

```shell
➜  samuha git:(main) ✗ npx nx serve api

> nx run api:serve:development

Build option outputFileName not set for api. Using fallback value of dist/apps/api/main.js.

> nx run api:build

[webpack-cli] Error: Cannot find module 'ajv/dist/compile/codegen'
Require stack:
- /Users/sagar/Sagar/samuha/node_modules/ajv-keywords/dist/definitions/typeof.js
- /Users/sagar/Sagar/samuha/node_modules/ajv-keywords/dist/keywords/typeof.js
- /Users/sagar/Sagar/samuha/node_modules/ajv-keywords/dist/keywords/index.js
- /Users/sagar/Sagar/samuha/node_modules/ajv-keywords/dist/index.js
- /Users/sagar/Sagar/samuha/node_modules/schema-utils/dist/validate.js
- /Users/sagar/Sagar/samuha/node_modules/schema-utils/dist/index.js
- /Users/sagar/Sagar/samuha/node_modules/copy-webpack-plugin/dist/index.js
- /Users/sagar/Sagar/samuha/node_modules/@nx/webpack/src/utils/create-copy-plugin.js
- /Users/sagar/Sagar/samuha/node_modules/@nx/webpack/index.js
- /Users/sagar/Sagar/samuha/apps/api/webpack.config.js
- /Users/sagar/Sagar/samuha/node_modules/webpack-cli/lib/webpack-cli.js
- /Users/sagar/Sagar/samuha/node_modules/webpack-cli/lib/bootstrap.js
- /Users/sagar/Sagar/samuha/node_modules/webpack-cli/bin/cli.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Module._load (node:internal/modules/cjs/loader:901:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (/Users/sagar/Sagar/samuha/node_modules/ajv-keywords/dist/definitions/typeof.js:3:19)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Module.require (node:internal/modules/cjs/loader:1115:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/sagar/Sagar/samuha/node_modules/ajv-keywords/dist/definitions/typeof.js',
    '/Users/sagar/Sagar/samuha/node_modules/ajv-keywords/dist/keywords/typeof.js',
    '/Users/sagar/Sagar/samuha/node_modules/ajv-keywords/dist/keywords/index.js',
    '/Users/sagar/Sagar/samuha/node_modules/ajv-keywords/dist/index.js',
    '/Users/sagar/Sagar/samuha/node_modules/schema-utils/dist/validate.js',
    '/Users/sagar/Sagar/samuha/node_modules/schema-utils/dist/index.js',
    '/Users/sagar/Sagar/samuha/node_modules/copy-webpack-plugin/dist/index.js',
    '/Users/sagar/Sagar/samuha/node_modules/@nx/webpack/src/utils/create-copy-plugin.js',
    '/Users/sagar/Sagar/samuha/node_modules/@nx/webpack/index.js',
    '/Users/sagar/Sagar/samuha/apps/api/webpack.config.js',
    '/Users/sagar/Sagar/samuha/node_modules/webpack-cli/lib/webpack-cli.js',
    '/Users/sagar/Sagar/samuha/node_modules/webpack-cli/lib/bootstrap.js',
    '/Users/sagar/Sagar/samuha/node_modules/webpack-cli/bin/cli.js'
  ]
}
Warning: run-commands command "webpack-cli build --node-env=production" exited with non-zero status code
 ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target build for project api (1s)

    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]

Build failed, waiting for changes to restart...
```


### Package Manager Version

_No response_

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

_No response_

--- a/e2e/angular-core/src/projects.test.ts
+++ b/e2e/angular-core/src/projects.test.ts
@@ -218,8 +218,7 @@ describe('Angular Projects', () => {
     removeFile(`${app1}/src/app/inline-template.component.ts`);
   }, 1000000);
 
-  // TODO(crystal, @jaysoo): enable this test when buildable libs work
-  xit('should build the dependent buildable lib and its child lib, as well as the app', async () => {
+  it('should build the dependent buildable lib and its child lib, as well as the app', async () => {
     // ARRANGE
     const buildableLib = uniq('buildlib1');
     const buildableChildLib = uniq('buildlib2');

--- a/packages/angular/src/builders/webpack-browser/webpack-browser.impl.ts
+++ b/packages/angular/src/builders/webpack-browser/webpack-browser.impl.ts
@@ -103,7 +103,7 @@ export function executeWebpackBrowserBuilder(
     switchMap(({ executeBrowserBuilder }) =>
       executeBrowserBuilder(delegateBuilderOptions, context as any, {
         webpackConfiguration: (baseWebpackConfig) => {
-          if (!buildLibsFromSource && delegateBuilderOptions.watch) {
+          if (!buildLibsFromSource) {
             const workspaceDependencies = dependencies
               .filter((dep) => !isNpmProject(dep.node))
               .map((dep) => dep.node.name);

--- a/packages/next/package.json
+++ b/packages/next/package.json
@@ -49,6 +49,7 @@
     "@nx/eslint": "file:../eslint",
     "@nx/react": "file:../react",
     "@nx/web": "file:../web",
+    "@nx/webpack": "file:../webpack",
     "@nx/workspace": "file:../workspace"
   },
   "publishConfig": {

--- a/packages/webpack/.eslintrc.json
+++ b/packages/webpack/.eslintrc.json
@@ -46,7 +46,8 @@
               "source-map-loader",
               "swc-loader",
               "@swc/core",
-              "ts-loader"
+              "ts-loader",
+              "ajv"
             ]
           }
         ]

--- a/packages/webpack/package.json
+++ b/packages/webpack/package.json
@@ -30,6 +30,7 @@
   },
   "dependencies": {
     "@babel/core": "^7.23.2",
+    "ajv": "^8.12.0",
     "autoprefixer": "^10.4.9",
     "babel-loader": "^9.1.2",
     "browserslist": "^4.21.4",

